src/ui-kit/experimental/hierarchical/hierarchical-tree-header/hierarchical-tree-header.component.ts

```diff
@@ -1,15 +1,61 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, Input, Output, EventEmitter, ViewChild, ElementRef } from '@angular/core';
+import { OptionsType } from '../../../../ui-kit/types';
+import { fromEvent } from 'rxjs/observable/fromEvent';
+import {
+  faThLarge,
+  faListUl,
+} from '@fortawesome/free-solid-svg-icons';
 
 @Component({
   selector: 'sam-hierarchical-tree-header',
   templateUrl: './hierarchical-tree-header.component.html',
   styleUrls: ['./hierarchical-tree-header.component.scss']
 })
-export class SamHierarchicalTreeHeaderComponent implements OnInit {
+export class SamHierarchicalTreeHeaderComponent {
+  faThLarge = faThLarge;
+  faListUl = faListUl;
 
-  constructor() { }
+  /**
+  * Options for the Dropdown
+  */
+  @Input() public options: OptionsType[];
+    /**
+  * Lable for the options dropdown
+  */
+  @Input() public label: string;
+    /**
+  * Whether Search should happned on click or keyup
+  */
+  @Input() public changeType: string = 'keyup';
+
+    /**
+  * Event emitted when level change is clicked
+  */
+  @Output() public selectedAgency = new EventEmitter<string>();
+    /**
+  * Event emitted when level change is clicked
+  */
+  @Output() public filterText = new EventEmitter<string>();
+
+  public selectModel: any;
+  @ViewChild('filter') filter: ElementRef;
 
   ngOnInit() {
+    fromEvent(this.filter.nativeElement, this.changeType)
+      .debounceTime(150)
```

Comment:
@yerramshilpa Can you make the debounceTime be a private variable for now? Let's not make it a public input yet, but that's something we might do in the future.